test/functional/ios/execute_driver_tests.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import textwrap
+import unittest
+
+from appium import webdriver
+from helper import desired_capabilities
+
+
+class ExecuteDriverTests(unittest.TestCase):
+    def setUp(self):
+        desired_caps = desired_capabilities.get_desired_capabilities('UICatalog.app.zip')
+        self.driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps)
+
+    def tearDown(self):
+        self.driver.quit()
+
+    def test_batch(self):
+        script = """
+            const status = await driver.status();
+            console.warn('warning message');
+            return status;
+        """
+
+        result = self.driver.execute_driver(script=textwrap.dedent(script))
+        assert(result.result['build'])
+        assert(result.logs['warn'] == ['warning message'])
+
+    def test_batch_combination_python_script(self):
+        script = """
+            console.warn('warning message');
+            const element = await driver.findElement('accessibility id', 'Buttons');
+            const rect = await driver.getElementRect(element.ELEMENT);
+            return [element, rect];
+        """
+
+        result = self.driver.execute_driver(script=textwrap.dedent(script))
+        r = result.result[0].rect
```

Comment:
I's say itis a bit confusing to have `result.result` mapping. What if we rename the retuned variable to `response` ?

Reply:
same below

Reply:
ah. Nice.  Yeah, I forgot re-naming it